### PR TITLE
Refactor how public and tag timelines are queried

### PR DESCRIPTION
**Pull request by Gargron**
Contains *some* minor behaviour changes:

- Excluded suspended accounts from public timelines (was not enforced previously)
- No longer show silenced user own posts in public timelines (silenced users can no longer make public posts anyway)
- Don't allow any/all/none filters in RSS and JSON-LD versions of the public hashtag resource
- Make JSON-LD version of public hashtag resource support min_id/since_id (consistently with outbox)

### CODE REVIEW

File: spec/models/public_feed_spec.rb [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    describe 'with an account passed in' do
+      before do
+        @account = Fabricate(:account)
+      end
+
+      subject { described_class.new(@account).get(20).map(&:id) }
+
+      it 'excludes statuses from accounts blocked by the account' do
+        blocked = Fabricate(:account)
+        Fabricate(:block, account: @account, target_account: blocked)
```
Comment (ThibG) [inline spec/models/public_feed_spec.rb:147]: `@account.block!(blocked)` would be easier to read

File: spec/models/public_feed_spec.rb [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        blocked_status = Fabricate(:status, account: blocked)
+
+        expect(subject).not_to include(blocked_status.id)
+      end
+
+      it 'excludes statuses from accounts who have blocked the account' do
+        blocked = Fabricate(:account)
+        Fabricate(:block, account: blocked, target_account: @account)
```
Comment (ThibG) [inline spec/models/public_feed_spec.rb:155]: `blocked.block!(@account)`… which makes me think the naming is a bit confusing

File: spec/models/public_feed_spec.rb [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        blocked_status = Fabricate(:status, account: blocked)
+
+        expect(subject).not_to include(blocked_status.id)
+      end
+
+      it 'excludes statuses from accounts muted by the account' do
+        muted = Fabricate(:account)
+        Fabricate(:mute, account: @account, target_account: muted)
```
Comment (ThibG) [inline spec/models/public_feed_spec.rb:163]: `@account.mute!(muted)`

File: app/models/public_feed.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class PublicFeed < Feed
+  def initialize(account, options = {})
```
Comment (ThibG) [inline app/models/public_feed.rb:4]: Documenting the `options` would be nice.

File: app/models/tag_feed.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class TagFeed < PublicFeed
+  LIMIT_PER_MODE = 4
+
+  def initialize(tag, account, options = {})
```
Comment (ThibG) [inline app/models/tag_feed.rb:6]: Documenting the `options` would be nice.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
I'm not sure I agree with the “Don't allow any/all/none filters in RSS and JSON-LD versions of the public hashtag resource” part, but otherwise it seems fine to me.

Minor remarks in inline comments.

Also, `HashtagQueryService` doesn't seem to be used anymore, but the PR doesn't remove its definition.

**Comment by Gargron** [comment]
> I'm not sure I agree with the “Don't allow any/all/none filters in RSS and JSON-LD versions of the public hashtag resource” part

Even if you discard that joining multiple tags is a more expensive operation, I don't think it's right for a JSON-LD resource to represent a complex union of different collections... It should be 1:1, in my opinion. Also nobody knows this feature even exists.

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**